test/Chainweb/Test/Pact/Utils.hs

```diff
@@ -846,6 +837,31 @@ toTxCreationTime (Time timespan) = case timeSpanToSeconds timespan of
 withPayloadDb :: (IO (PayloadDb HashMapCas) -> TestTree) -> TestTree
 withPayloadDb = withResource newPayloadDb (\_ -> return ())
 
+
+-- | 'MemPoolAccess' that delegates all calls to the contents of provided `IORef`.
+delegateMemPoolAccess :: IORef MemPoolAccess -> MemPoolAccess
+delegateMemPoolAccess r = MemPoolAccess
+  { mpaGetBlock = \a b c d -> call mpaGetBlock $ \f -> f a b c d
+  , mpaSetLastHeader = \a -> call mpaSetLastHeader ($ a)
+  , mpaProcessFork = \a -> call mpaProcessFork ($ a)
+  , mpaBadlistTx = \a -> call mpaBadlistTx ($ a)
+  }
+  where
+    call :: (MemPoolAccess -> f) -> (f -> IO a) -> IO a
+    call f g = readIORef r >>= g . f
+
+-- | use a "delegate" which you can dynamically reset/modify
+withDelegateMempool
+  :: (IO (IORef MemPoolAccess, MemPoolAccess) -> TestTree)
+  -> TestTree
+withDelegateMempool = withResource start (const mempty)
```

Comment:
Isn't `(const mempty :: a -> IO ()) == (mempty :: a -> IO())`?
 ```suggestion
withDelegateMempool = withResource start mempty
```

Reply:
Went around and fixed this elsewhere in the module too :)